src/Parsers/FunctionParameterValuesVisitor.cpp

```diff
@@ -64,15 +67,20 @@ class FunctionParameterValuesVisitor
                         parameter_values[identifier->name()] = convertFieldToString(cast_literal->value);
                     }
                 }
+                else
+                {
+                    ASTPtr res = evaluateConstantExpressionOrIdentifierAsLiteral(expression_list->children[1], context);
```

Comment:
IMHO it's better to avoid using the old infrastructure.

Reply:
do you mean `evaluateConstantExpressionOrIdentifierAsLiteral` function ? Could you please suggest what cna be used instead in new analyzer?

Reply:
Yes, I mean this function. It uses `TreeRewriter` and `ExpressionAnalyzer` to evaluate the constant. It's an excellent question if we need to be able to execute some AST to get a constant value.
However, I see many usages of `evaluateConstantExpressionOrIdentifierAsLiteral`. So, I suspect we need to rewrite it, but it's better to do that in a separate PR. It shouldn't be a blocker for your PR.